### DISCUSSION TITLE
Is there a callback for when the build is complete?

### DISCUSSION THREAD

**Discussion by SharpIceX** [hexojs/hexo#5500] [Question / Answer]
I'm currently working on a searchable database of all pages, but I didn't find any callback API provided by Hexo for when the build is complete. I don't know if there is a better implementation other than Node's own `process.on('exit' () => {}) callback`.

This is my current code (testing):
``` javascript
'use strict';

var hexo = hexo || {};
const searchDatabase = [];

if (hexo.env.cmd !== 'server') {
	hexo.extend.filter.register(
		'after_render:html',
		function (htmlContent, data) {
			searchDatabase.push(htmlContent);
		},
		1002,
	);
	process.on('exit', function () {
		console.log(searchDatabase.length);
	});
}

```

**Comment by SukkaW** [comment]
You don't need to wait for the build to finish before generating a searchable database.

Checkout the implementation here: https://github.com/next-theme/hexo-generator-searchdb

**Reply by SharpIceX** [reply]
> 您无需等待构建完成即可生成可搜索数据库。
> 
> 在此处查看实现： [ https://github.com/next-theme/hexo-generator-searchdb ](https://github.com/next-theme/hexo-generator-searchdb)

But using `hexo.extend.generator.register` can't get all pages
